hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMultiMapProxy.java

```diff
@@ -91,6 +105,98 @@ public ClientMultiMapProxy(String serviceName, String name, ClientContext contex
         super(serviceName, name, context);
     }
 
+    @Override
+    public CompletionStage<Void> putAllAsync(@Nonnull Map<? extends K, Collection<? extends V>> m) {
+        InternalCompletableFuture<Void> future = new InternalCompletableFuture<>();
+        Map<Data, Data> dataMap = new HashMap<>();
+
+        for (Map.Entry e : m.entrySet()) {
+            Collection<Data> dataCollection = CollectionUtil
+                    .objectToDataCollection(((Collection<? extends V>) e.getValue()),
+                            getSerializationService());
+
+            dataMap.put(toData(e.getKey()), toData(new DataCollection(dataCollection)));
+        }
+        putAllInternal(dataMap, future);
+        return future;
+    }
+
+    @Override
+    public CompletionStage<Void> putAllAsync(@Nonnull K key, Collection<? extends V> value) {
+        InternalCompletableFuture<Void> future = new InternalCompletableFuture<>();
+        Map<Data, Data> dataMap = new HashMap<>();
+
+        Collection<Data> dataCollection = CollectionUtil
+                .objectToDataCollection(value, getSerializationService());
+        dataMap.put(toData(key), toData(new DataCollection(dataCollection)));
+        putAllInternal(dataMap, future);
+        return future;
+    }
+
+
+    @SuppressWarnings({"checkstyle:cyclomaticcomplexity", "checkstyle:npathcomplexity", "checkstyle:methodlength"})
+    private void putAllInternal(@Nonnull Map<Data, Data> map,
+                                @Nullable InternalCompletableFuture<Void> future) {
```

Comment:
I could not find a usage of this future with a null value. 

@vbekiaris Are we planning to implement sync versions of these methods ? 
If there is no plan, can we make this `@nonnull` , and remove the null checks inside the method? Same goes for the MultiMapProxyImpl as well.

Reply:
Quoting from https://github.com/hazelcast/hazelcast/issues/9079#issuecomment-584523865 :

> ... So, instead of getAll and putAll, you can add getAllAsync and putAllAsync.

There is no concrete plan for implementing the synchronous version of these methods, unless there is interest / a PR from the community. So I would agree to have the `future == null` paths removed from this PR.